### PR TITLE
Reorganized Overlay and Thinpool sections

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1809486

Preview build: http://file.rdu.redhat.com/~ahardin/April-6-2020/choosing-graph-driver/scaling_performance/optimizing_storage.html#choosing-a-graph-driver

### CODE REVIEW

**Comment by fgiloux** [comment]
@ahardin-rh thank you for looking into that. A few comments from my side:
Under "Choosing a graph driver"
Using a loop device can affect performance issues => Using a loop device can affect performance
Under "For Overlay"
point 2: With overlayfs to have imagefs on a different logical volume you would need CONTAINER_ROOT_LV_NAME and possibly CONTAINER_ROOT_LV_MOUNT_PATH
See the following links: https://github.com/projectatomic/container-storage-setup/blob/master/container-storage-setup.conf#L118-L120
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/managing_containers/managing_storage_with_docker_formatted_containers#using_the_overlay_graph_driver
point 3 important section
Red Hat strongly recommends using the overlay2 storage driver in thin-pool mode for production workloads.
=> This can be removed. "thin pool" is commonly used with devicemapper to differentiate direct-lvm, which is using a thin pool from loop-lvm, which is using sparse files.
OverlayFS is a union mount filesystem. You don't need to choose between direct-lvm and loop-lvm

Regards,

Frédéric

**Comment by ahardin-rh** [comment]
@fgiloux Thank you! This is now updated.

@liangxia  Are you the correct contact to provide QE review? Thank you!

**Comment by liangxia** [comment]
@qinpingli PTAL

File: scaling_performance/optimizing_storage.adoc [hunk 1]
```diff
@@ -293,7 +281,51 @@ If using CRI-O, specify `STORAGE_DRIVER=overlay`.
 With CRI-O, the default `overlay` storage driver uses the `overlay2` optimizations.
 ====
 +
-. If you had a separate disk drive dedicated to Docker storage (for example,
+With OverlayFS, if you want to have `imagefs` on a different logical volume,
+then you must set `CONTAINER_ROOT_LV_NAME` and `CONTAINER _ROOT_LV_MOUNT_PATH`.
```
Comment (wjiangjay) [inline scaling_performance/optimizing_storage.adoc:285]: additional space here

**Comment by qinpingli** [comment]
@wjiangjay will give a review from QE side.

**Comment by wjiangjay** [comment]
LGTM.
overlay2:
```
# cat /etc/sysconfig/docker-storage-setup 
STORAGE_DRIVER=overlay2
CONTAINER_ROOT_LV_NAME="container-root_lv"
CONTAINER_ROOT_LV_MOUNT_PATH="/var/lib/docker"
# lsblk
NAME                        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
vda                         252:0    0   40G  0 disk 
├─vda1                      252:1    0  500M  0 part /boot
└─vda2                      252:2    0  9.5G  0 part 
  ├─rhel-root               253:0    0  8.8G  0 lvm  /
  └─rhel-container--root_lv 253:1    0  292M  0 lvm  /var/lib/docker
vdb                         252:16   0   20G  0 disk 
# docker info | egrep -i 'storage|pool|space|filesystem'
Storage Driver: overlay2
 Backing Filesystem: xfs
```

devicemapper:
```
[root@basiton ~]# cat /etc/sysconfig/docker-storage-setup                                                                                                                                                          
STORAGE_DRIVER=devicemapper
DEVS=/dev/vdb

# lsblk /dev/vdb                                                                                                                                                                                   
NAME                             MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
vdb                              252:16   0  20G  0 disk 
└─vdb1                           252:17   0  20G  0 part 
  ├─docker_vg-docker--pool_tmeta 253:2    0  24M  0 lvm  
  │ └─docker_vg-docker--pool     253:4    0   8G  0 lvm  
  └─docker_vg-docker--pool_tdata 253:3    0   8G  0 lvm  
    └─docker_vg-docker--pool     253:4    0   8G  0 lvm  
# docker info | egrep -i 'storage|pool|space|filesystem'                                                                                                                              
Storage Driver: devicemapper
 Pool Name: docker_vg-docker--pool
 Pool Blocksize: 524.3 kB
 Backing Filesystem: xfs
 Data Space Used: 19.92 MB
 Data Space Total: 8.535 GB
 Data Space Available: 8.515 GB
 Metadata Space Used: 2.556 MB
 Metadata Space Total: 25.17 MB
 Metadata Space Available: 22.61 MB
 Thin Pool Minimum Free Space: 853.5 MB
```

**Pull request merged**